### DISCUSSION TITLE
Beta 2: Server init hook and how to render html attribute

### DISCUSSION THREAD

**Discussion by rolanday** [solidjs/solid-start#1164] [Q&A]
Hoping I'll have better luck here than discord, which has been a disappointment lately getting questions answered. I'd be happy to pay for Solid support (in addition to my monthly contrib) if that's what it takes to get an answer as I value my time, and this is last ditch effort to get answer to how-to's before throwing in the towel on Solid. Very basic stuff, that I've had zero issues implementing in alternate frameworks (qwik, svelte, remix, next).

Beta 2

1. How to render an attribute into html root element (in entry-server.ts)) w/ beta2. In Beta 1, I was using root.data alongside root.tsx, but that's no longer an option w/ beta 2.
https://discord.com/channels/722131463138705510/1187172047076479056/1187172047076479056

2. Where to hook into start-up so I can async run init code before start starts. Use case is commonplace -- grab secrets from secrets manager which are needed to handle processing requests.
https://discord.com/channels/722131463138705510/1187172491391668244/1187172491391668244

Any help much appreciated. Sorry to be a grinch, but it's a bit frustrating to have these go unanswered when I see a flurry of responses from maintainers in other areas, and these were re-posts to support due to lack of response in community after 3 or 4 days now.

**Comment by ryansolid** [comment]
Sorry for the delay. At the time of releasing Beta 2 we were still figuring this out. The answer to the 2nd question is there is middleware that can run async code before Start, starts up: https://start.solidjs.com/advanced/middleware. Here stuff can be added to `event.locals`.

The first question is a little trickier. As for rendering into the server entry, technically you still can, but the document gets rendered only on the server and outside of the app entry. So you need read stuff off the `event` and if you need things to update dynamically later in the client then you would need to use `createEffect` or the like.
